daemon/cluster/cluster.go

```diff
@@ -1108,7 +1108,7 @@ func (c *Cluster) CreateService(s types.ServiceSpec, encodedAuth string) (*apity
 		if err != nil {
 			logrus.Warnf("unable to pin image %s to digest: %s", ctnr.Image, err.Error())
 			resp.Warnings = append(resp.Warnings, fmt.Sprintf("unable to pin image %s to digest: %s", ctnr.Image, err.Error()))
-		} else {
+		} else if ctnr.Image != digestImage {
```

Comment:
I see your point. I had deliberately kept the log message on `create` but didn't consider how it can look misleading.

Perhaps it's reasonable to add a separate message when no pinning happens, such as `creating service with image digest busybox:latest@sha256:29f5d56d12684887bdfa50dcd29fc31eea4aaf4ad3bec43daf19026a7ce69912`, and leave the existing message for the default case.

Either way, I don't feel too strongly about it.